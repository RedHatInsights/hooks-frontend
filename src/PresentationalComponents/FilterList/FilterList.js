import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
    Checkbox,
    List,
    ListItem
} from '@patternfly/react-core';
import _ from 'lodash';

export class FilterList extends Component {
    static propTypes = {
        apps: PropTypes.object.isRequired,
        selectedAppEventTypes: PropTypes.shape({
            appIds: PropTypes.array,
            eventTypeIds: PropTypes.array,
            levelIds: PropTypes.array
        })
    };

    removeOrAddId = (array: [], id) =>
        array.indexOf(id) !== -1 ? array.filter((currentId) => currentId !== id) : [ ...array, id ]

    eventTypeCheckboxChange = (eventTypeId) =>
        this.props.selectedAppEventTypes.eventTypeIds = this.removeOrAddId(this.props.selectedAppEventTypes.eventTypeIds, eventTypeId)

    appCheckboxChange = (appId) =>
        this.props.selectedAppEventTypes.appIds = this.removeOrAddId(this.props.selectedAppEventTypes.appIds, appId)

    levelCheckboxChange = (levelId) =>
        this.props.selectedAppEventTypes.levelIds = this.removeOrAddId(this.props.selectedAppEventTypes.levelIds, levelId)

    isEventTypeEnabled = (eventTypeId) =>
        _.includes(this.props.selectedAppEventTypes.eventTypeIds, parseInt(eventTypeId))

    isAppEnabled = (appId) =>
        _.includes(this.props.selectedAppEventTypes.appIds, parseInt(appId))

    isLevelEnabled = (levelId) =>
        _.includes(this.props.selectedAppEventTypes.levelIds, parseInt(levelId))

    renderLevel = (level) =>
        level.attributes ?
            <ListItem key={ `level-${ level.id}` }>
                <Checkbox id={ `level-check-${ level.id}` }
                    data-event-type-id={ level.id }
                    label={ level.attributes.title }
                    aria-label={ level.attributes.title }
                    onChange={ () => this.levelCheckboxChange(level.id) }
                    defaultChecked={ this.isLevelEnabled(level.id) } />
            </ListItem> : '';

    renderLevels = (levels) => {
        let levelsArray = _.values(levels);
        return levelsArray.length > 0 ?
            <List>
                { levelsArray.map((level) =>
                    this.renderLevel(level)
                ) }
            </List> : '';
    }

    eventTypesListItem = (eventType) =>
        eventType.attributes ?
            <ListItem key={ `event-type-${ eventType.id}` }>
                <Checkbox id={ `event-type-check-${ eventType.id}` }
                    data-event-type-id={ eventType.id }
                    label={ eventType.attributes.name }
                    aria-label={ eventType.attributes.name }
                    onChange={ () => this.eventTypeCheckboxChange(eventType.id) }
                    defaultChecked={ this.isEventTypeEnabled(eventType.id) } />
                { this.renderLevels(eventType.levels) }
            </ListItem> : '';

    eventTypesList = (eventTypes) => {
        let eventTypesArray = _.values(eventTypes);
        return eventTypesArray.length > 0 ?
            <List>
                { eventTypesArray.map((eventType) =>
                    this.eventTypesListItem(eventType)
                ) }
            </List> : '';
    }

    render() {
        const apps = _.values(this.props.apps);

        return <List>
            { apps.map((app) =>
                <ListItem key={ `app-${ app.id }` }>
                    <Checkbox id={ `app-check-${ app.id}` }
                        data-event-type-id={ app.id }
                        label={ app.attributes.name }
                        aria-label={ app.attributes.name }
                        onChange={ () => this.appCheckboxChange(app.id) }
                        defaultChecked={ this.isAppEnabled(app.id) } />
                    { this.eventTypesList(app.eventTypes, app.id) }
                </ListItem>
            ) }
        </List>;
    }
}

export default FilterList;
