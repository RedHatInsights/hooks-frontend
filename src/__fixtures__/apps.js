const apps = {
    data: [
        {
            id: 1,
            type: 'app',
            attributes: {
                name: 'seed-app-0'
            },
            relationships: {
                event_types: {
                    data: []
                }
            }
        },
        {
            id: 2,
            type: 'app',
            attributes: {
                name: 'seed-app-1'
            },
            relationships: {
                event_types: {
                    data: []
                }
            }
        },
        {
            id: 3,
            type: 'app',
            attributes: {
                name: 'seed-app-2'
            },
            relationships: {
                event_types: {
                    data: [
                        {
                            id: 1,
                            type: 'event_type'
                        },
                        {
                            id: 2,
                            type: 'event_type'
                        }
                    ]
                }
            }
        },
        {
            id: 4,
            type: 'app',
            attributes: {
                name: 'seed-app-3'
            },
            relationships: {
                event_types: {
                    data: [
                        {
                            id: 3,
                            type: 'event_type'
                        },
                        {
                            id: 4,
                            type: 'event_type'
                        }
                    ]
                }
            }
        },
        {
            id: 5,
            type: 'app',
            attributes: {
                name: 'seed-app-4'
            },
            relationships: {
                event_types: {
                    data: []
                }
            }
        },
        {
            id: 6,
            type: 'app',
            attributes: {
                name: 'seed-app-5'
            },
            relationships: {
                event_types: {
                    data: [
                        {
                            id: 5,
                            type: 'event_type'
                        },
                        {
                            id: 6,
                            type: 'event_type'
                        },
                        {
                            id: 7,
                            type: 'event_type'
                        }
                    ]
                }
            }
        },
        {
            id: 7,
            type: 'app',
            attributes: {
                name: 'seed-app-6'
            },
            relationships: {
                event_types: {
                    data: []
                }
            }
        },
        {
            id: 8,
            type: 'app',
            attributes: {
                name: 'seed-app-7'
            },
            relationships: {
                event_types: {
                    data: [
                        {
                            id: 8,
                            type: 'event_type'
                        },
                        {
                            id: 9,
                            type: 'event_type'
                        },
                        {
                            id: 10,
                            type: 'event_type'
                        }
                    ]
                }
            }
        },
        {
            id: 9,
            type: 'app',
            attributes: {
                name: 'seed-app-8'
            },
            relationships: {
                event_types: {
                    data: [
                        {
                            id: 11,
                            type: 'event_type'
                        },
                        {
                            id: 12,
                            type: 'event_type'
                        },
                        {
                            id: 13,
                            type: 'event_type'
                        },
                        {
                            id: 14,
                            type: 'event_type'
                        }
                    ]
                }
            }
        },
        {
            id: 10,
            type: 'app',
            attributes: {
                name: 'seed-app-9'
            },
            relationships: {
                event_types: {
                    data: [
                        {
                            id: 15,
                            type: 'event_type'
                        },
                        {
                            id: 16,
                            type: 'event_type'
                        },
                        {
                            id: 17,
                            type: 'event_type'
                        },
                        {
                            id: 18,
                            type: 'event_type'
                        }
                    ]
                }
            }
        }
    ],
    included: [
        {
            id: 1,
            type: 'event_type',
            attributes: {
                title: 'seed-type-0',
                name: 'seed-type-0'
            },
            relationships: {
                levels: {
                    data: [
                        {
                            id: 1,
                            type: 'level'
                        }
                    ]
                }
            }
        },
        {
            id: 2,
            type: 'event_type',
            attributes: {
                title: 'seed-type-1',
                name: 'seed-type-1'
            },
            relationships: {
                levels: {
                    data: [
                        {
                            id: 2,
                            type: 'level'
                        }
                    ]
                }
            }
        },
        {
            id: 1,
            type: 'level',
            attributes: {
                title: 'level-0'
            }
        },
        {
            id: 2,
            type: 'level',
            attributes: {
                title: 'level-0'
            }
        },
        {
            id: 3,
            type: 'event_type',
            attributes: {
                title: 'seed-type-0',
                name: 'seed-type-0'
            },
            relationships: {
                levels: {
                    data: [
                        {
                            id: 3,
                            type: 'level'
                        },
                        {
                            id: 4,
                            type: 'level'
                        },
                        {
                            id: 5,
                            type: 'level'
                        }
                    ]
                }
            }
        },
        {
            id: 4,
            type: 'event_type',
            attributes: {
                title: 'seed-type-1',
                name: 'seed-type-1'
            },
            relationships: {
                levels: {
                    data: [
                        {
                            id: 6,
                            type: 'level'
                        },
                        {
                            id: 7,
                            type: 'level'
                        },
                        {
                            id: 8,
                            type: 'level'
                        }
                    ]
                }
            }
        },
        {
            id: 3,
            type: 'level',
            attributes: {
                title: 'level-0'
            }
        },
        {
            id: 4,
            type: 'level',
            attributes: {
                title: 'level-1'
            }
        },
        {
            id: 5,
            type: 'level',
            attributes: {
                title: 'level-2'
            }
        },
        {
            id: 6,
            type: 'level',
            attributes: {
                title: 'level-0'
            }
        },
        {
            id: 7,
            type: 'level',
            attributes: {
                title: 'level-1'
            }
        },
        {
            id: 8,
            type: 'level',
            attributes: {
                title: 'level-2'
            }
        },
        {
            id: 5,
            type: 'event_type',
            attributes: {
                title: 'seed-type-0',
                name: 'seed-type-0'
            },
            relationships: {
                levels: {
                    data: [
                        {
                            id: 9,
                            type: 'level'
                        },
                        {
                            id: 10,
                            type: 'level'
                        },
                        {
                            id: 11,
                            type: 'level'
                        },
                        {
                            id: 12,
                            type: 'level'
                        }
                    ]
                }
            }
        },
        {
            id: 6,
            type: 'event_type',
            attributes: {
                title: 'seed-type-1',
                name: 'seed-type-1'
            },
            relationships: {
                levels: {
                    data: [
                        {
                            id: 13,
                            type: 'level'
                        },
                        {
                            id: 14,
                            type: 'level'
                        },
                        {
                            id: 15,
                            type: 'level'
                        },
                        {
                            id: 16,
                            type: 'level'
                        }
                    ]
                }
            }
        },
        {
            id: 7,
            type: 'event_type',
            attributes: {
                title: 'seed-type-2',
                name: 'seed-type-2'
            },
            relationships: {
                levels: {
                    data: [
                        {
                            id: 17,
                            type: 'level'
                        },
                        {
                            id: 18,
                            type: 'level'
                        },
                        {
                            id: 19,
                            type: 'level'
                        },
                        {
                            id: 20,
                            type: 'level'
                        }
                    ]
                }
            }
        },
        {
            id: 9,
            type: 'level',
            attributes: {
                title: 'level-0'
            }
        },
        {
            id: 10,
            type: 'level',
            attributes: {
                title: 'level-1'
            }
        },
        {
            id: 11,
            type: 'level',
            attributes: {
                title: 'level-2'
            }
        },
        {
            id: 12,
            type: 'level',
            attributes: {
                title: 'level-3'
            }
        },
        {
            id: 13,
            type: 'level',
            attributes: {
                title: 'level-0'
            }
        },
        {
            id: 14,
            type: 'level',
            attributes: {
                title: 'level-1'
            }
        },
        {
            id: 15,
            type: 'level',
            attributes: {
                title: 'level-2'
            }
        },
        {
            id: 16,
            type: 'level',
            attributes: {
                title: 'level-3'
            }
        },
        {
            id: 17,
            type: 'level',
            attributes: {
                title: 'level-0'
            }
        },
        {
            id: 18,
            type: 'level',
            attributes: {
                title: 'level-1'
            }
        },
        {
            id: 19,
            type: 'level',
            attributes: {
                title: 'level-2'
            }
        },
        {
            id: 20,
            type: 'level',
            attributes: {
                title: 'level-3'
            }
        },
        {
            id: 8,
            type: 'event_type',
            attributes: {
                title: 'seed-type-0',
                name: 'seed-type-0'
            },
            relationships: {
                levels: {
                    data: [
                        {
                            id: 21,
                            type: 'level'
                        },
                        {
                            id: 22,
                            type: 'level'
                        },
                        {
                            id: 23,
                            type: 'level'
                        },
                        {
                            id: 24,
                            type: 'level'
                        }
                    ]
                }
            }
        },
        {
            id: 9,
            type: 'event_type',
            attributes: {
                title: 'seed-type-1',
                name: 'seed-type-1'
            },
            relationships: {
                levels: {
                    data: [
                        {
                            id: 25,
                            type: 'level'
                        },
                        {
                            id: 26,
                            type: 'level'
                        },
                        {
                            id: 27,
                            type: 'level'
                        },
                        {
                            id: 28,
                            type: 'level'
                        }
                    ]
                }
            }
        },
        {
            id: 10,
            type: 'event_type',
            attributes: {
                title: 'seed-type-2',
                name: 'seed-type-2'
            },
            relationships: {
                levels: {
                    data: [
                        {
                            id: 29,
                            type: 'level'
                        },
                        {
                            id: 30,
                            type: 'level'
                        },
                        {
                            id: 31,
                            type: 'level'
                        },
                        {
                            id: 32,
                            type: 'level'
                        }
                    ]
                }
            }
        },
        {
            id: 21,
            type: 'level',
            attributes: {
                title: 'level-0'
            }
        },
        {
            id: 22,
            type: 'level',
            attributes: {
                title: 'level-1'
            }
        },
        {
            id: 23,
            type: 'level',
            attributes: {
                title: 'level-2'
            }
        },
        {
            id: 24,
            type: 'level',
            attributes: {
                title: 'level-3'
            }
        },
        {
            id: 25,
            type: 'level',
            attributes: {
                title: 'level-0'
            }
        },
        {
            id: 26,
            type: 'level',
            attributes: {
                title: 'level-1'
            }
        },
        {
            id: 27,
            type: 'level',
            attributes: {
                title: 'level-2'
            }
        },
        {
            id: 28,
            type: 'level',
            attributes: {
                title: 'level-3'
            }
        },
        {
            id: 29,
            type: 'level',
            attributes: {
                title: 'level-0'
            }
        },
        {
            id: 30,
            type: 'level',
            attributes: {
                title: 'level-1'
            }
        },
        {
            id: 31,
            type: 'level',
            attributes: {
                title: 'level-2'
            }
        },
        {
            id: 32,
            type: 'level',
            attributes: {
                title: 'level-3'
            }
        },
        {
            id: 11,
            type: 'event_type',
            attributes: {
                title: 'seed-type-0',
                name: 'seed-type-0'
            },
            relationships: {
                levels: {
                    data: []
                }
            }
        },
        {
            id: 12,
            type: 'event_type',
            attributes: {
                title: 'seed-type-1',
                name: 'seed-type-1'
            },
            relationships: {
                levels: {
                    data: []
                }
            }
        },
        {
            id: 13,
            type: 'event_type',
            attributes: {
                title: 'seed-type-2',
                name: 'seed-type-2'
            },
            relationships: {
                levels: {
                    data: []
                }
            }
        },
        {
            id: 14,
            type: 'event_type',
            attributes: {
                title: 'seed-type-3',
                name: 'seed-type-3'
            },
            relationships: {
                levels: {
                    data: []
                }
            }
        },
        {
            id: 15,
            type: 'event_type',
            attributes: {
                title: 'seed-type-0',
                name: 'seed-type-0'
            },
            relationships: {
                levels: {
                    data: [
                        {
                            id: 33,
                            type: 'level'
                        },
                        {
                            id: 34,
                            type: 'level'
                        },
                        {
                            id: 35,
                            type: 'level'
                        },
                        {
                            id: 36,
                            type: 'level'
                        }
                    ]
                }
            }
        },
        {
            id: 16,
            type: 'event_type',
            attributes: {
                title: 'seed-type-1',
                name: 'seed-type-1'
            },
            relationships: {
                levels: {
                    data: [
                        {
                            id: 37,
                            type: 'level'
                        },
                        {
                            id: 38,
                            type: 'level'
                        },
                        {
                            id: 39,
                            type: 'level'
                        },
                        {
                            id: 40,
                            type: 'level'
                        }
                    ]
                }
            }
        },
        {
            id: 17,
            type: 'event_type',
            attributes: {
                title: 'seed-type-2',
                name: 'seed-type-2'
            },
            relationships: {
                levels: {
                    data: [
                        {
                            id: 41,
                            type: 'level'
                        },
                        {
                            id: 42,
                            type: 'level'
                        },
                        {
                            id: 43,
                            type: 'level'
                        },
                        {
                            id: 44,
                            type: 'level'
                        }
                    ]
                }
            }
        },
        {
            id: 18,
            type: 'event_type',
            attributes: {
                title: 'seed-type-3',
                name: 'seed-type-3'
            },
            relationships: {
                levels: {
                    data: [
                        {
                            id: 45,
                            type: 'level'
                        },
                        {
                            id: 46,
                            type: 'level'
                        },
                        {
                            id: 47,
                            type: 'level'
                        },
                        {
                            id: 48,
                            type: 'level'
                        }
                    ]
                }
            }
        },
        {
            id: 33,
            type: 'level',
            attributes: {
                title: 'level-0'
            }
        },
        {
            id: 34,
            type: 'level',
            attributes: {
                title: 'level-1'
            }
        },
        {
            id: '35',
            type: 'level',
            attributes: {
                title: 'level-2'
            }
        },
        {
            id: 36,
            type: 'level',
            attributes: {
                title: 'level-3'
            }
        },
        {
            id: 37,
            type: 'level',
            attributes: {
                title: 'level-0'
            }
        },
        {
            id: 38,
            type: 'level',
            attributes: {
                title: 'level-1'
            }
        },
        {
            id: 39,
            type: 'level',
            attributes: {
                title: 'level-2'
            }
        },
        {
            id: 40,
            type: 'level',
            attributes: {
                title: 'level-3'
            }
        },
        {
            id: 41,
            type: 'level',
            attributes: {
                title: 'level-0'
            }
        },
        {
            id: 42,
            type: 'level',
            attributes: {
                title: 'level-1'
            }
        },
        {
            id: 43,
            type: 'level',
            attributes: {
                title: 'level-2'
            }
        },
        {
            id: 44,
            type: 'level',
            attributes: {
                title: 'level-3'
            }
        },
        {
            id: 45,
            type: 'level',
            attributes: {
                title: 'level-0'
            }
        },
        {
            id: 46,
            type: 'level',
            attributes: {
                title: 'level-1'
            }
        },
        {
            id: 47,
            type: 'level',
            attributes: {
                title: 'level-2'
            }
        },
        {
            id: 48,
            type: 'level',
            attributes: {
                title: 'level-3'
            }
        }
    ],
    meta: {
        total: 10,
        per_page: 10,
        page: 1
    }
};

export default apps;
