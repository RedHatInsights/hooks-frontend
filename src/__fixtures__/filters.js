const filters = {
    data: [
        {
            id: 13,
            type: 'filter',
            attributes: {
                enabled: true
            },
            relationships: {
                apps: {
                    data: [
                        {
                            id: 1,
                            type: 'app'
                        },
                        {
                            id: 3,
                            type: 'app'
                        },
                        {
                            id: 4,
                            type: 'app'
                        },
                        {
                            id: 8,
                            type: 'app'
                        },
                        {
                            id: '4',
                            type: 'app'
                        }
                    ]
                },
                event_types: {
                    data: [
                        {
                            id: 1,
                            type: 'event_type'
                        },
                        {
                            id: 3,
                            type: 'event_type'
                        },
                        {
                            id: 8,
                            type: 'event_type'
                        }
                    ]
                },
                levels: {
                    data: [
                        {
                            id: 1,
                            type: 'level'
                        },
                        {
                            id: '4',
                            type: 'level'
                        },
                        {
                            id: '22',
                            type: 'level'
                        }
                    ]
                },
                endpoints: {
                    data: [
                        {
                            id: 36,
                            type: 'endpoint'
                        }
                    ]
                }
            }
        }
    ],
    meta: {
        total: 1,
        per_page: 10,
        page: 1
    }
};

export default filters;
