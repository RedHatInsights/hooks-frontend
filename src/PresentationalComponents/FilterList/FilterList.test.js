import { FilterList } from './FilterList';

import apps from '../../__fixtures__/apps';
import filters from '../../__fixtures__/filters';

describe('FilterList', () => {
    it('expect to render', () => {
        const selectedAppEventTypes = {
            appIds: [ 1 ],
            eventTypeIds: [ 11 ]
        };
        const wrapper = shallow(
            <FilterList filters={ filters } apps={ apps.data } selectedAppEventTypes={ selectedAppEventTypes } />
        );

        expect(wrapper.render()).toMatchSnapshot();
    });
});
