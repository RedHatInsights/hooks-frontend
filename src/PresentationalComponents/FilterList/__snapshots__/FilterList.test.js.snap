// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterList expect to render 1`] = `
<div
  class="pf-l-stack pf-m-gutter"
>
  <div
    class="pf-l-stack__item"
  >
    <div
      class="pf-c-check"
    >
      <input
        aria-invalid="false"
        aria-label="Seed App 0"
        checked=""
        class="pf-c-check__input"
        data-event-type-id="1"
        id="app-check-1"
        type="checkbox"
      />
      <label
        class="pf-c-check__label"
        for="app-check-1"
      >
        <strong>
          Seed App 0
        </strong>
      </label>
    </div>
    <div
      style="margin-top:1em;margin-left:1em"
    >
      <div
        class="pf-c-radio"
      >
        <input
          aria-invalid="false"
          checked=""
          class="pf-c-radio__input"
          id="app-1-radio-all"
          name="app-1-event-type-radio"
          type="radio"
          value="all"
        />
        <label
          class="pf-c-radio__label"
          for="app-1-radio-all"
        >
          All event types
        </label>
      </div>
    </div>
  </div>
  <div
    class="pf-l-stack__item"
  >
    <div
      class="pf-c-check"
    >
      <input
        aria-invalid="false"
        aria-label="Seed App 1"
        class="pf-c-check__input"
        data-event-type-id="2"
        id="app-check-2"
        type="checkbox"
      />
      <label
        class="pf-c-check__label"
        for="app-check-2"
      >
        <strong>
          Seed App 1
        </strong>
      </label>
    </div>
  </div>
  <div
    class="pf-l-stack__item"
  >
    <div
      class="pf-c-check"
    >
      <input
        aria-invalid="false"
        aria-label="Seed App 2"
        checked=""
        class="pf-c-check__input"
        data-event-type-id="3"
        id="app-check-3"
        type="checkbox"
      />
      <label
        class="pf-c-check__label"
        for="app-check-3"
      >
        <strong>
          Seed App 2
        </strong>
      </label>
    </div>
    <div
      style="margin-top:1em;margin-left:1em"
    >
      <div
        class="pf-c-radio"
      >
        <input
          aria-invalid="false"
          class="pf-c-radio__input"
          id="app-3-radio-all"
          name="app-3-event-type-radio"
          type="radio"
          value="all"
        />
        <label
          class="pf-c-radio__label"
          for="app-3-radio-all"
        >
          All event types
        </label>
      </div>
      <div
        class="pf-c-radio"
      >
        <input
          aria-invalid="false"
          checked=""
          class="pf-c-radio__input"
          id="app-3-radio-selected"
          name="app-3-event-type-radio"
          type="radio"
          value="selected-only"
        />
        <label
          class="pf-c-radio__label"
          for="app-3-radio-selected"
        >
          Only selected event types
        </label>
      </div>
      <ul
        class="pf-c-list"
      >
        <li>
          <div
            class="pf-c-check"
          >
            <input
              aria-invalid="false"
              aria-label="Seed Type 0"
              checked=""
              class="pf-c-check__input"
              data-event-type-id="1"
              id="event-type-check-1"
              type="checkbox"
            />
            <label
              class="pf-c-check__label"
              for="event-type-check-1"
            >
              Seed Type 0
            </label>
          </div>
          <ul
            class="pf-c-list"
          >
            <li>
              <div
                class="pf-c-check"
              >
                <input
                  aria-invalid="false"
                  aria-label="level-0"
                  checked=""
                  class="pf-c-check__input"
                  data-event-type-id="1"
                  id="level-check-1"
                  type="checkbox"
                />
                <label
                  class="pf-c-check__label"
                  for="level-check-1"
                >
                  level-0
                </label>
              </div>
            </li>
          </ul>
        </li>
        <li>
          <div
            class="pf-c-check"
          >
            <input
              aria-invalid="false"
              aria-label="Seed Type 1"
              class="pf-c-check__input"
              data-event-type-id="2"
              id="event-type-check-2"
              type="checkbox"
            />
            <label
              class="pf-c-check__label"
              for="event-type-check-2"
            >
              Seed Type 1
            </label>
          </div>
        </li>
      </ul>
    </div>
  </div>
  <div
    class="pf-l-stack__item"
  >
    <div
      class="pf-c-check"
    >
      <input
        aria-invalid="false"
        aria-label="Seed App 3"
        checked=""
        class="pf-c-check__input"
        data-event-type-id="4"
        id="app-check-4"
        type="checkbox"
      />
      <label
        class="pf-c-check__label"
        for="app-check-4"
      >
        <strong>
          Seed App 3
        </strong>
      </label>
    </div>
    <div
      style="margin-top:1em;margin-left:1em"
    >
      <div
        class="pf-c-radio"
      >
        <input
          aria-invalid="false"
          class="pf-c-radio__input"
          id="app-4-radio-all"
          name="app-4-event-type-radio"
          type="radio"
          value="all"
        />
        <label
          class="pf-c-radio__label"
          for="app-4-radio-all"
        >
          All event types
        </label>
      </div>
      <div
        class="pf-c-radio"
      >
        <input
          aria-invalid="false"
          checked=""
          class="pf-c-radio__input"
          id="app-4-radio-selected"
          name="app-4-event-type-radio"
          type="radio"
          value="selected-only"
        />
        <label
          class="pf-c-radio__label"
          for="app-4-radio-selected"
        >
          Only selected event types
        </label>
      </div>
      <ul
        class="pf-c-list"
      >
        <li>
          <div
            class="pf-c-check"
          >
            <input
              aria-invalid="false"
              aria-label="Seed Type 0"
              checked=""
              class="pf-c-check__input"
              data-event-type-id="3"
              id="event-type-check-3"
              type="checkbox"
            />
            <label
              class="pf-c-check__label"
              for="event-type-check-3"
            >
              Seed Type 0
            </label>
          </div>
          <ul
            class="pf-c-list"
          >
            <li>
              <div
                class="pf-c-check"
              >
                <input
                  aria-invalid="false"
                  aria-label="level-0"
                  class="pf-c-check__input"
                  data-event-type-id="3"
                  id="level-check-3"
                  type="checkbox"
                />
                <label
                  class="pf-c-check__label"
                  for="level-check-3"
                >
                  level-0
                </label>
              </div>
            </li>
            <li>
              <div
                class="pf-c-check"
              >
                <input
                  aria-invalid="false"
                  aria-label="level-1"
                  checked=""
                  class="pf-c-check__input"
                  data-event-type-id="4"
                  id="level-check-4"
                  type="checkbox"
                />
                <label
                  class="pf-c-check__label"
                  for="level-check-4"
                >
                  level-1
                </label>
              </div>
            </li>
            <li>
              <div
                class="pf-c-check"
              >
                <input
                  aria-invalid="false"
                  aria-label="level-2"
                  class="pf-c-check__input"
                  data-event-type-id="5"
                  id="level-check-5"
                  type="checkbox"
                />
                <label
                  class="pf-c-check__label"
                  for="level-check-5"
                >
                  level-2
                </label>
              </div>
            </li>
          </ul>
        </li>
        <li>
          <div
            class="pf-c-check"
          >
            <input
              aria-invalid="false"
              aria-label="Seed Type 1"
              class="pf-c-check__input"
              data-event-type-id="4"
              id="event-type-check-4"
              type="checkbox"
            />
            <label
              class="pf-c-check__label"
              for="event-type-check-4"
            >
              Seed Type 1
            </label>
          </div>
        </li>
      </ul>
    </div>
  </div>
  <div
    class="pf-l-stack__item"
  >
    <div
      class="pf-c-check"
    >
      <input
        aria-invalid="false"
        aria-label="Seed App 4"
        class="pf-c-check__input"
        data-event-type-id="5"
        id="app-check-5"
        type="checkbox"
      />
      <label
        class="pf-c-check__label"
        for="app-check-5"
      >
        <strong>
          Seed App 4
        </strong>
      </label>
    </div>
  </div>
  <div
    class="pf-l-stack__item"
  >
    <div
      class="pf-c-check"
    >
      <input
        aria-invalid="false"
        aria-label="Seed App 5"
        class="pf-c-check__input"
        data-event-type-id="6"
        id="app-check-6"
        type="checkbox"
      />
      <label
        class="pf-c-check__label"
        for="app-check-6"
      >
        <strong>
          Seed App 5
        </strong>
      </label>
    </div>
  </div>
  <div
    class="pf-l-stack__item"
  >
    <div
      class="pf-c-check"
    >
      <input
        aria-invalid="false"
        aria-label="Seed App 6"
        class="pf-c-check__input"
        data-event-type-id="7"
        id="app-check-7"
        type="checkbox"
      />
      <label
        class="pf-c-check__label"
        for="app-check-7"
      >
        <strong>
          Seed App 6
        </strong>
      </label>
    </div>
  </div>
  <div
    class="pf-l-stack__item"
  >
    <div
      class="pf-c-check"
    >
      <input
        aria-invalid="false"
        aria-label="Seed App 7"
        checked=""
        class="pf-c-check__input"
        data-event-type-id="8"
        id="app-check-8"
        type="checkbox"
      />
      <label
        class="pf-c-check__label"
        for="app-check-8"
      >
        <strong>
          Seed App 7
        </strong>
      </label>
    </div>
    <div
      style="margin-top:1em;margin-left:1em"
    >
      <div
        class="pf-c-radio"
      >
        <input
          aria-invalid="false"
          class="pf-c-radio__input"
          id="app-8-radio-all"
          name="app-8-event-type-radio"
          type="radio"
          value="all"
        />
        <label
          class="pf-c-radio__label"
          for="app-8-radio-all"
        >
          All event types
        </label>
      </div>
      <div
        class="pf-c-radio"
      >
        <input
          aria-invalid="false"
          checked=""
          class="pf-c-radio__input"
          id="app-8-radio-selected"
          name="app-8-event-type-radio"
          type="radio"
          value="selected-only"
        />
        <label
          class="pf-c-radio__label"
          for="app-8-radio-selected"
        >
          Only selected event types
        </label>
      </div>
      <ul
        class="pf-c-list"
      >
        <li>
          <div
            class="pf-c-check"
          >
            <input
              aria-invalid="false"
              aria-label="Seed Type 0"
              checked=""
              class="pf-c-check__input"
              data-event-type-id="8"
              id="event-type-check-8"
              type="checkbox"
            />
            <label
              class="pf-c-check__label"
              for="event-type-check-8"
            >
              Seed Type 0
            </label>
          </div>
          <ul
            class="pf-c-list"
          >
            <li>
              <div
                class="pf-c-check"
              >
                <input
                  aria-invalid="false"
                  aria-label="level-0"
                  class="pf-c-check__input"
                  data-event-type-id="21"
                  id="level-check-21"
                  type="checkbox"
                />
                <label
                  class="pf-c-check__label"
                  for="level-check-21"
                >
                  level-0
                </label>
              </div>
            </li>
            <li>
              <div
                class="pf-c-check"
              >
                <input
                  aria-invalid="false"
                  aria-label="level-1"
                  checked=""
                  class="pf-c-check__input"
                  data-event-type-id="22"
                  id="level-check-22"
                  type="checkbox"
                />
                <label
                  class="pf-c-check__label"
                  for="level-check-22"
                >
                  level-1
                </label>
              </div>
            </li>
            <li>
              <div
                class="pf-c-check"
              >
                <input
                  aria-invalid="false"
                  aria-label="level-2"
                  class="pf-c-check__input"
                  data-event-type-id="23"
                  id="level-check-23"
                  type="checkbox"
                />
                <label
                  class="pf-c-check__label"
                  for="level-check-23"
                >
                  level-2
                </label>
              </div>
            </li>
            <li>
              <div
                class="pf-c-check"
              >
                <input
                  aria-invalid="false"
                  aria-label="level-3"
                  class="pf-c-check__input"
                  data-event-type-id="24"
                  id="level-check-24"
                  type="checkbox"
                />
                <label
                  class="pf-c-check__label"
                  for="level-check-24"
                >
                  level-3
                </label>
              </div>
            </li>
          </ul>
        </li>
        <li>
          <div
            class="pf-c-check"
          >
            <input
              aria-invalid="false"
              aria-label="Seed Type 1"
              class="pf-c-check__input"
              data-event-type-id="9"
              id="event-type-check-9"
              type="checkbox"
            />
            <label
              class="pf-c-check__label"
              for="event-type-check-9"
            >
              Seed Type 1
            </label>
          </div>
        </li>
        <li>
          <div
            class="pf-c-check"
          >
            <input
              aria-invalid="false"
              aria-label="Seed Type 2"
              class="pf-c-check__input"
              data-event-type-id="10"
              id="event-type-check-10"
              type="checkbox"
            />
            <label
              class="pf-c-check__label"
              for="event-type-check-10"
            >
              Seed Type 2
            </label>
          </div>
        </li>
      </ul>
    </div>
  </div>
  <div
    class="pf-l-stack__item"
  >
    <div
      class="pf-c-check"
    >
      <input
        aria-invalid="false"
        aria-label="Seed App 8"
        class="pf-c-check__input"
        data-event-type-id="9"
        id="app-check-9"
        type="checkbox"
      />
      <label
        class="pf-c-check__label"
        for="app-check-9"
      >
        <strong>
          Seed App 8
        </strong>
      </label>
    </div>
  </div>
  <div
    class="pf-l-stack__item"
  >
    <div
      class="pf-c-check"
    >
      <input
        aria-invalid="false"
        aria-label="Seed App 9"
        class="pf-c-check__input"
        data-event-type-id="10"
        id="app-check-10"
        type="checkbox"
      />
      <label
        class="pf-c-check__label"
        for="app-check-10"
      >
        <strong>
          Seed App 9
        </strong>
      </label>
    </div>
  </div>
</div>
`;
